File: Most-Recursion/Questions/generateSubsets.js
function subsets(arr,i,curr,powerSet){
    if(i == arr.length){
        powerSet.push([...curr])
        return;
    }

    // leave the current Element
    subsets(arr,i+1,curr,powerSet);

    // add the current element
    curr.push(arr[i]);

    // take the current element
    subsets(arr,i+1,curr,powerSet)

    // remove the current element 
    curr.pop();// back tracking

}
let arr = [1,2,3];
let powerSet = [];
subsets(arr,0,[],powerSet);
console.log(powerSet);

File: Most-Recursion/Questions/generateSubsets.py
def subSet(arr,i,curr,powerSet):
    if(i == len(arr)):
        powerSet.append(curr.copy())# Corrected to copy the current subset
        return

     # Exclude the current element and move to the next
    subSet(arr,i+1,curr,powerSet)

    # Include the current element
    curr.append(arr[i])
    subSet(arr,i+1,curr,powerSet)

    # Backtrack to remove the current element
    curr.pop()

lst = [1,2,3]
powerSet = []
subSet(lst,0,[],powerSet)
print(powerSet)